themes/bootstrap3/templates/Recommend/WorldCatIdentities.phtml

```diff
@@ -14,17 +14,30 @@
           <dd><?=$this->transEsc('Related Subjects')?>:</dd>
           <?php $j = 0; foreach ($subjects as $subj): ?>
             <?php if (++$j == 3): ?>
-              <dd id="more-WCIdents<?=$i?>"><a href="#" onclick="moreFacets('WCIdents<?=$i?>'); return false;"><?=$this->transEsc('more')?> ...</a></dd>
+              <dd id="more-WCIdents<?=$i?>"><a href="#"><?=$this->transEsc('more')?> ...</a></dd>
               <span class="hidden WCIdents<?=$i?>">
             <?php endif; ?>
             <dd>&bull; <a href="<?=$this->url('search-results')?>?lookfor=%22<?=urlencode($subj)?>%22&amp;type=Subject"><?=$this->escapeHtml($subj)?></a></dd>
           <?php endforeach; ?>
-          <?php if ($j > 2): ?><dd><a href="#" onclick="lessFacets('WCIdents<?=$i?>'); return false;"><?=$this->transEsc('less')?> ...</a></dd></span><?php endif; ?>
+          <?php if ($j > 2): ?><dd id="less-WCIdents">><a href="#"><?=$this->transEsc('less')?> ...</a></dd></span><?php endif; ?>
           </dl>
         <?php endif; ?>
         </dd>
       <?php endforeach; ?>
-      <?php if ($i > 3): ?><dd><a href="#" onclick="lessFacets('WCIdents'); return false;"><?=$this->transEsc('less')?> ...</a></dd></span><?php endif; ?>
+      <?php if ($i > 3): ?><dd id="less-WCIdents"><a href="#"><?=$this->transEsc('less')?> ...</a></dd></span><?php endif; ?>
     </dl>
   </div>
 <?php endif; ?>
+<?php
+$script = <<<JS
+$('#more-WCIdents a').click(function(e) {
+    moreFacets('WCIdents');
```

Comment:
This template calls moreFacets/lessFacets, but these have been removed from common.js. Unless I'm missing something, I think that means there is a problem here.

Reply:
I forget to fix this when I refactored facets js last time... now it should be fixed

Reply:
I think a little more work may be needed, for two reasons:

1.) If you load the WorldCatIdentities recommendation module on a page that has no other facet recommendation modules, facets.js won't load, and then the functionality won't work. It is probably worth including the JS file in the template; it will be automatically deduplicated, so I don't think there's any negative side effect to doing that.

2.) I haven't tested yet, but I'm concerned that calling registerMoreLessFacetsEventHandlers() multiple times may register duplicate event handlers, which might have weird side effects. It might be wise to `.unbind('click')` before setting up the event handlers.

If you don't have time for this, I can do more testing/adjustments tomorrow or Monday!

Reply:
Ad 1) it is the same with top facets, I added facets.js to both templates

Ad 2) yes, it is probably better to be on the safer side... register function now first remove the handler. As the handlers body is written now, it should have no side effects, but it could change in the future... we should not rely on that the implementation is always save and free of side effects...